src/components/ColorPickerInput/ColorPickerInput.tsx

```diff
@@ -0,0 +1,112 @@
+import {Palette} from '@gravity-ui/icons';
+import {Button, Flex, Icon, TextInput, TextInputProps} from '@gravity-ui/uikit';
+import {useTranslation} from 'next-i18next';
+import React, {ChangeEventHandler, useCallback, useRef, useState} from 'react';
+
+import {block} from '../../utils';
+
+import './ColorPickerInput.scss';
+import {ColorPreview} from './ColorPreview';
+import {NativeColorPicker} from './NativeColorPicker';
+import {hexRegexp, parseRgbStringToHex, rgbRegexp, rgbaRegexp} from './utils';
+
+const b = block('color-picker');
+
+export interface ColorPickerInputProps {
+    defaultValue: string;
+    name?: string;
+    value?: string;
+    onChange?: (color: string) => void;
+    errorMessage?: string;
+}
+
+export const ColorPickerInput = ({
+    name,
+    value,
+    onChange: onChangeExternal,
+    defaultValue,
+    errorMessage,
+}: ColorPickerInputProps) => {
+    const {t} = useTranslation('component');
+
+    const [color, setColor] = useState<string>(defaultValue);
+    const [inputValue, setInputValue] = useState<string>(defaultValue);
+    const [validationError, setValidationError] = useState<TextInputProps['validationState']>();
+
+    const colorInputRef = useRef<HTMLInputElement>(null);
+
+    const managedValue = value || inputValue;
+
+    const onChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+        (event) => {
+            const newValue = event.target.value.replaceAll(' ', '');
+            onChangeExternal?.(newValue);
+            setInputValue(newValue);
+            setValidationError(undefined);
+
+            if (
+                !newValue ||
+                new RegExp(hexRegexp, 'g').test(newValue) ||
+                new RegExp(rgbaRegexp, 'g').test(newValue)
+            ) {
+                setColor(newValue);
+
+                return;
+            }
+
+            if (new RegExp(rgbRegexp, 'g').test(newValue)) {
+                const hexColor = parseRgbStringToHex(newValue);
+
+                setColor(hexColor);
+                return;
+            }
+        },
+        [onChangeExternal],
+    );
+
+    const onNativeInputChange: ChangeEventHandler<HTMLInputElement> = useCallback((e) => {
+        const newValue = e.target.value.toUpperCase();
+
+        setColor(newValue);
+        setInputValue(newValue);
```

Comment:
It seems that the validation status needs to be reset here.

Reply:
I will add it in next task (or u can add it by yourself)